core/icons/comment_icon.ts

```diff
@@ -0,0 +1,326 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Block} from '../block.js';
+import {BlockSvg} from '../block_svg.js';
+import {COMMENT_TYPE} from './icon_types.js';
+import {Coordinate} from '../utils.js';
+import * as dom from '../utils/dom.js';
+import * as eventUtils from '../events/utils.js';
+import {Icon} from './icon.js';
+import {IHasBubble} from '../interfaces/i_has_bubble.js';
+import {ISerializable} from '../interfaces/i_serializable.js';
+import {Rect} from '../utils/rect.js';
+import * as registry from './registry.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import {TextBubble} from '../bubbles/text_bubble.js';
+import {TextInputBubble} from '../bubbles/textinput_bubble.js';
+import {WorkspaceSvg} from '../workspace_svg.js';
+
+export class CommentIcon extends Icon implements IHasBubble, ISerializable {
+  /** The type string used to identify this icon. */
+  static readonly TYPE = COMMENT_TYPE;
+
+  /**
+   * The weight this icon has relative to other icons. Icons with more positive
+   * weight values are rendered farther toward the end of the block.
+   */
+  static readonly WEIGHT = 3;
+
+  /** The size of this icon in workspace-scale units. */
+  private readonly SIZE = 17;
+
+  /** The default width in workspace-scale units of the text input bubble. */
+  private readonly DEFAULT_BUBBLE_WIDTH = 160;
+
+  /** The default height in workspace-scale units of the text input bubble. */
+  private readonly DEFAULT_BUBBLE_HEIGHT = 80;
```

Comment:
nit: I don't think non-static properties of a class meet the requirements to be constant case: "Global: Only symbols declared on the module level, static fields of module level classes, and values of module level enums, may use CONST_CASE."